imports/ui/components/profile/profile_page.js

```diff
@@ -40,5 +43,11 @@ Template.profilePage.helpers({
              user.profile.facebook || user.profile.snapchat ||
              user.profile.soundcloud;
     } else return false;
+  },
+  posts: function () {
+    var username = FlowRouter.getParam('username');
+    var user = Meteor.users.findOne({username: username});
+    var i = user._id;
+    return Posts.find({userId: i});
```

Comment:
Even after limiting the publication just to a single user's posts, keep the `userId` query parameter in case other Posts are available in the collection for whatever reason. I'd also specify a reverse-chronological sort to return to the template.